internal/xds/translator/translator.go

```diff
@@ -532,9 +532,28 @@ func (t *Translator) addRouteToRouteConfig(
 			}
 
 			var err error
+			var clusters []*clusterv3.Cluster
+			switch {
+			case len(httpRoute.BackendExtensionRefs) > 0:
+				clusters, err = processExtensionBackendPostHook(xdsRoute, vHost, httpRoute, t.ExtensionManager)
```

Comment:
we cannot call the hook twice, server may not garuntee idempotency + 2 calls will increase latency / route convergence

we'll have to reuse L503

Reply:
Changed